controllers/templatesync/template_sync.go

```diff
@@ -1193,11 +1193,29 @@ func generatePendingMsg(dependencyFailures []depclient.ObjectIdentifier) string
 
 func overrideRemediationAction(instance *policiesv1.Policy, tObjectUnstructured *unstructured.Unstructured) {
 	// override RemediationAction only when it is set on parent
-	if instance.Spec.RemediationAction == "" {
-		return
-	}
+	// or when a policy is informonly
+	if tObjectUnstructured.GroupVersionKind().Group != utils.GConstraint &&
+		tObjectUnstructured.GroupVersionKind().Group != utils.GvkConstraintTemplate.Group {
+		if spec, ok := tObjectUnstructured.Object["spec"]; ok {
+			specObject, ok := spec.(map[string]interface{})
+			if ok {
+				if remediationAction, ok := specObject["remediationAction"]; ok {
+					_, ok := remediationAction.(string)
+					if ok {
```

Comment:
All of the logic looks right. But it's not as easy to read as I would prefer. My personal preference is to reduce indentation when possible.

Since this is a lot of checks, one trick would be to consider what should be done when a check fails: in this case, it should just return without changing any settings. So the first two checks could be re-written like this:

```go
spec, ok := tObjectUnstructured.Object["spec"]
if !ok {
    return // template didn't have a spec
}

specObject, ok := spec.(map[string]interface{})
if !ok {
    return // template's spec wasn't a map
}
...
```

Maybe in general this function should return an error (those comments indicate what might go into the error message), but it hasn't historically, so I don't think there's a need to add that now.

Reply:
Alternately, you could save lines by using the `if ___; ok {` thing every time, at least to be consistent.

```suggestion
		if spec, ok := tObjectUnstructured.Object["spec"]; ok {
			if specObject, ok := spec.(map[string]interface{}); ok {
				if remediationAction, ok := specObject["remediationAction"]; ok {
					if remAction, ok := remediationAction.(string); ok {
```

Reply:
With regards to this, I was trying to be consistent in the way the assertions were already being made in the code. Should I change all the comma-ok assertions in this function so that they are each evaluated in one line?

Reply:
Ah, now I see that the existing case had the one-liner for the lookup, but had it separated for the type assertion. So this was consistently following the inconsistency 😅 

Honestly, I'll accept any form you're happy with (as long as it's not much crazier than what is already in the PR). As I said, the logic looks right. De-tangling the test cases was my main reason for requesting changes on the PR, otherwise I would've approved it with these as optional suggestions.